onnx/backend/test/case/node/cast.py

```diff
@@ -25,18 +25,26 @@ def export():  # type: () -> None
             ('FLOAT16', 'DOUBLE'),
             ('DOUBLE', 'FLOAT'),
             ('DOUBLE', 'FLOAT16'),
+            ('FLOAT', 'STRING'),
+            ('STRING', 'FLOAT'),
         ]
 
         for from_type, to_type in test_cases:
-            input = np.random.random_sample(shape).astype(
-                TENSOR_TYPE_TO_NP_TYPE[getattr(TensorProto, from_type)])
+            if 'STRING' != from_type:
+                input = np.random.random_sample(shape).astype(
+                    TENSOR_TYPE_TO_NP_TYPE[getattr(TensorProto, from_type)])
+
+                output = input.astype(TENSOR_TYPE_TO_NP_TYPE[getattr(TensorProto, to_type)])
+            else:
+                input = np.array([['0.47892547', '0.48033667', '0.49968487', '0.81910545'],
```

Comment:
Do you have tests for inf and nan? The spec also needs to mention the string literals of them.